Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1481,12 +1481,14 @@ void FrameLoader::loadFrameRequest(FrameLoadRequest&& request, Event* event, Ref
     if (loadType == FrameLoadType::Reload) {
         if (m_frame->document() && m_frame->document()->settings().navigationAPIEnabled()) {
             if (RefPtr window = frame->document()->window()) {
-                RefPtr<SerializedScriptValue> stateObject;
-                if (RefPtr currentItem = history().currentItem())
-                    stateObject = currentItem->navigationAPIStateObject();
-                RefPtr sourceElement = event ? dynamicDowncast<Element>(event->target()) : nullptr;
-                if (!dispatchNavigateEvent(url, loadType, request.downloadAttribute(), request.navigationHistoryBehavior(), false, formState.get(), stateObject.get(), sourceElement.get()))
-                    return;
+                if (!request.skipNavigateEvent()) {
```

Comment:
It seems like kind of an anti pattern to require all callers of dispatchNavigateEvent to check this boolean.  It seems like it would make the code a bit cleaner to just pass in a bool indicating whether the navigation event was dispatched when scheduled.  That would also make it so that future callers won't forget it.